### ISSUE TITLE
SSL_ERROR_RX_RECORD_TOO_LONG

### ISSUE THREAD

**Issue by StarScream159** [pyload/pyload#2493]
Hey guys,

When I enable SSL from the menu with my cert and key I get the following error when I try to load the site over HTTPS.

Am I doing something wrong? I've confirmed that the paths are correct and that the cert/key is good (it worked for FTP).

**Comment by Velociraptor85** [comment]
@StarScream159 can you send us some more data? Debug logs and a more precise error description.
Where do you get the error, at which point...

**Comment by StarScream159** [comment]
I can try, but this is going to be hard as nothing that I see in pyLoad logs.

I'll point form here:
- my OS (it's the minimal one)
> [root@centbox ~]# cat /etc/redhat-release
> CentOS Linux release 7.2.1511 (Core)
> [root@centbox ~]#

- running:
```
[root@centbox freenas]# python /opt/pyload/pyLoadCore.py
06.07.2016 03:30:57 INFO      Starting pyLoad 0.4.9
06.07.2016 03:30:57 INFO      Using home directory: /opt/pyload/.pyload
06.07.2016 03:30:57 DEBUG     Remote activated: True
06.07.2016 03:30:58 DEBUG     created index of plugins
06.07.2016 03:30:58 INFO      Activated plugins: EventMapper, MergeFiles, UnSkipOnFail, UpdateManager, UserAgentSwitcher, XFileSharing
06.07.2016 03:30:58 INFO      Deactivate plugins: AndroidPhoneNotify, AntiStandby, AntiVirus, BypassCaptcha, Captcha9Kw, CaptchaBrotherhood, Checksum, ClickNLoad, DeathByCaptcha, DeleteFinished, DownloadScheduler, ExpertDecoders, ExternalScripts, ExtractArchive, HotFolder, IRC, ImageTyperz, JustPremium, LinkdecrypterComHook, LogMarker, MultiHome, PushBullet, PushOver, RestartFailed, SkipRev, TransmissionRPC, WindowsPhoneNotify, XMPP
06.07.2016 03:30:58 INFO      Downloadtime: True
06.07.2016 03:30:58 INFO      Using SSL ThriftBackend
06.07.2016 03:30:58 INFO      Starting ThriftBackend: 0.0.0.0:7227
06.07.2016 03:30:58 INFO      Starting threaded webserver: 0.0.0.0:8000
06.07.2016 03:30:58 INFO      Free space: 491.59 GiB
06.07.2016 03:30:59 INFO      Activating Accounts...
```
I drop out the account information because I don't think that would apply. I do have some so if you need the details reply.

**Comment by GammaC0de** [comment]
@StarScream159 Have you followed the instructions in #2000?

**Comment by StarScream159** [comment]
@GammaC0de Yes, I followed the instructions to install way back when. Since then I've updated the UploadManager.py file with a different version that was posted here that fixed an issue with false update notices.

To test though, I backed up my current version and redownloaded the one from #2000. I tested SSL but it did the same thing. It also produced an error in my log, so I reverted back to the version I have:
`21.07.2016 07:45:06 INFO      pyLoad is up and running
21.07.2016 07:45:53 ERROR     Error importing UpdateManager: No module named Hook
`

When I attempt to load pyLoad (over https) in Chrome I get the following:

> This site can’t provide a secure connection
> 192.168.2.25 sent an invalid response.
> Try:
> Reloading the page
> Learn more about this problem.
> ERR_SSL_PROTOCOL_ERROR

And in firefox I get the following:

> Secure Connection Failed

An error occurred during a connection to 192.168.2.25:8000. SSL received a record that exceeded the maximum permissible length. Error code: SSL_ERROR_RX_RECORD_TOO_LONG

    The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
    Please contact the website owners to inform them of this problem.

Learn more…

**Comment by GammaC0de** [comment]
>21.07.2016 07:45:53 ERROR     Error importing UpdateManager: No module named Hook

Your pyload installation is broken.
Try updating your `UpdateManager.py` from https://raw.githubusercontent.com/pyload/pyload/stable/module/plugins/hooks/UpdateManager.py and see what happens

**Comment by Velociraptor85** [comment]
@StarScream159 any news on the error? did the suggestion from @GammaC0de help you?

**Comment by StarScream159** [comment]
@Velociraptor85 thanks. No, nothing has changed. I've just been accessing it over http. I will try the UpdateManager GammaC0de linked to tonight and see if that will change anything. I really suspect I am missing libraries for my distro.

I'll update the ticket this evening.

**Comment by BKPepe** [comment]
Have same issue as the OP.
```
root@Debian-NAS:/home/pepe/pyload# python pyLoadCore.py
02.06.2017 15:48:09 INFO      Starting pyLoad 0.4.9
02.06.2017 15:48:09 INFO      Using home directory: /home/pepe/.pyload
02.06.2017 15:48:09 INFO      Activated plugins: ClickNLoad, EventMapper, ExternalScripts, ExtractArchive, MergeFiles, UnSkipOnFail, UpdateManager, UserAgentSwitcher, XFileSharing
02.06.2017 15:48:09 INFO      Deactivate plugins: AndroidPhoneNotify, AntiStandby, AntiVirus, BypassCaptcha, Captcha9Kw, CaptchaBrotherhood, Checksum, CloudFlareDdos, DeathByCaptcha, DeleteFinished, DownloadScheduler, ExpertDecoders, HotFolder, IRC, ImageTyperz, JustPremium, LinkFilter, LinkdecrypterComHook, LogMarker, MultiHome, PushBullet, PushOver, RestartFailed, SkipRev, TransmissionRPC, WindowsPhoneNotify, XMPP
02.06.2017 15:48:09 INFO      Downloadtime: True
02.06.2017 15:48:10 INFO      Starting threaded webserver: 0.0.0.0:8000
02.06.2017 15:48:10 INFO      Free space: 1.69 TiB
02.06.2017 15:48:10 INFO      Activating Accounts...
02.06.2017 15:48:10 INFO      Activating Plugins...
02.06.2017 15:48:10 INFO      ADDON XFileSharing: Auto-discover new hosters
02.06.2017 15:48:10 INFO      ADDON XFileSharing: Auto-discover new crypters
02.06.2017 15:48:10 INFO      pyLoad is up and running
02.06.2017 15:48:11 INFO      ADDON UpdateManager: pyLoad is up to date!
02.06.2017 15:48:11 INFO      ADDON UpdateManager: All plugins are up to date!
02.06.2017 15:48:16 INFO      ADDON ClickNLoad: Proxy listening on 0.0.0.0:9666
```
My OS latest version of Debian (and I'm also running OMV on it)
Kernel: 
```
root@Debian-NAS:~# uname -a
Linux Debian-NAS 4.9.0-0.bpo.3-amd64 #1 SMP Debian 4.9.25-1~bpo8+1 (2017-05-19) x86_64 GNU/Linux
```

Accessing through https:
![pyload_https](https://cloud.githubusercontent.com/assets/4096468/26728965/4745037a-47ac-11e7-8542-5de62193f934.jpg)

Of course I followed #2000 , but no go. 
**I use threaded server**
Accessing w/o https works.

**Comment by StarScream159** [comment]
I just update by install via https://github.com/pyload/pyload/archive/stable.zip and tried this all again. I get the same error still. I am also using the Threaded web server.

Testing the other webserver options is a good idea. I did the same, the results are below:

**Threaded**:
As above, nothing in the logs. Firefox reports SSL_ERROR_RX_RECORD_TOO_LONG.

**Builtin**:
The log reports:
```
192.168.2.17 - - [02/Jun/2017 19:13:45] code 400, message Bad HTTP/0.9 request type ('\x16\x03\x01\x00\xa3\x01\x00\x00\x9f\x03\x03')
```

**Fastcgi**:
The log reports:
```
192.168.2.17 - - [02/Jun/2017 19:18:50] code 400, message Bad HTTP/0.9 request type ('\x16\x03\x01\x00\xa3\x01\x00\x00\x9f\x03\x03\xff\x83')
```

**lightweight**
The log reports:
```
192.168.2.17 - - [02/Jun/2017 19:19:47] code 400, message Bad request syntax ('\x16\x03\x01\x00\xa3\x01\x00\x00\x9f\x03\x03\xa1\xb2\xb7\x0e)\xba\x83\x7f\x04\x04n\x1c\xd3\x18\xf9,\xf7\xb8\xe9\x84EJ\x9a\xd3n\xd8\xb9:\x94dP\xc8\x00\x00\x1e\xc0+\xc0/\xcc\xa9\xcc\xa8\xc0,\xc00\xc0')
```

**Comment by StarScream159** [comment]
Oh my... I'm a dummy!

I have solved this issue. I had a similar error today when doing something unrelated to pyLoad whereby I was trying to run a nginx proxy for something, however, I had the configuration incorrect and was attempting to pass un-encrypted traffic over the HTTPS port with SSL disabled and no certs passed to the conf. So the browser expects an encrypted connection, but is receiving a non encrypted one.

This led me to think maybe pyLoad was attempting to communicate via HTTP over the HTTPS port and the cert wasn't being correctly assigned.

Hunting around I found the setting 'Use HTTPS:' (under General -> Webinterface) was set to OFF, even though the setting 'Activated' (under General -> SSL) was set to ON. So this was indeed making the HTTP request on the specified port reply with a NON encrypted connection without the certs.

Once the settings were changed, and a restart preformed (this is important) everything started to work correctly.

So really, **this was a PEBKAC** issue the whole time. However, maybe it would be good to add a warning message for when you set 'Activated' for General -> SSL but still have 'Use HTTPS:' set to false. Or move the 'Use HTTPS' setting to the SSL screen? _Not sure._

At least the issue is fixed. I'll close this now.
